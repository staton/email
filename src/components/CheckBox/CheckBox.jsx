import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {MdCheckBoxOutlineBlank, MdCheckBox} from 'react-icons/md';

const propTypes = {
    isChecked: PropTypes.bool,
    onClick: PropTypes.func
};

const defaultProps = {
    isChecked: false,
    onClick: () => {}
};

class CheckBox extends Component {

    constructor() {
        super();

        this.handleClicked = this.handleClicked.bind(this);
    }
	render() {
		return (
            <div 
                className={this.getClassName()}
                onClick={this.handleClicked}>
            {
                (this.props.isChecked) 
                ?   <MdCheckBox /> 
                :   <MdCheckBoxOutlineBlank />
            }
            </div>
		);
    }

    /**
     * Gets the class name(s) for this component.
     * @returns {string} The class name(s).
     */
    getClassName() {
        let className = 'CheckBox';

        if (this.props.isChecked) {
            className += ' CheckBox--checked';
        }

        return className;
    }

    /**
     * Called when the checkbox is clicked.
     * @param {object} e The event.
     */
    handleClicked(e) {
        e.preventDefault();
        e.stopPropagation();
        this.props.onClick();
    }

}

CheckBox.propTypes = propTypes;
CheckBox.defaultProps = defaultProps;

export default CheckBox;