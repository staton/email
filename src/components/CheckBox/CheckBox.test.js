import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import CheckBox from './CheckBox';

describe('<CheckBox />', () => {

    it('renders correctly when unchecked', () => {
        const wrapper = shallow(<CheckBox isChecked={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders correctly when checked', () => {
        const wrapper = shallow(<CheckBox isChecked={true} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('doesn\'t have the ..--checked class when not checked', () => {
        const wrapper = shallow(<CheckBox isChecked={false} />);
        expect(wrapper.find('.CheckBox--checked').length).toEqual(0);
    });

    it('has the ..--checked class when checked', () => {
        const wrapper = shallow(<CheckBox isChecked={true} />);
        expect(wrapper.find('.CheckBox--checked').length).toEqual(1);
    });
    
    it('handles click events', () => {
        const onClick = jest.fn();
        const wrapper = mount(<CheckBox onClick={onClick} />);
        let button = wrapper.find('.CheckBox');

        button.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    });
    
});