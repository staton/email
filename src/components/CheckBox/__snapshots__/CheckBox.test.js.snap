// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> renders correctly when checked 1`] = `
<div
  className="CheckBox CheckBox--checked"
  onClick={[Function]}
>
  <Component />
</div>
`;

exports[`<CheckBox /> renders correctly when unchecked 1`] = `
<div
  className="CheckBox"
  onClick={[Function]}
>
  <Component />
</div>
`;
